pkg/util/resize/configspec_devices.go

```diff
@@ -25,16 +26,633 @@ func compareHardwareDevices(
+	}
+
+	{
+		exp, cur := selectDevsByType[*vimtypes.VirtualPointingDevice](expectedDevices, currentDevices)
+		dc := zipVirtualDevicesOfType(MatchVirtualPointingDevice, exp, cur)
+		deviceChanges = append(deviceChanges, dc...)
+	}
+
+	{
+		exp, cur := selectDevsByType[*vimtypes.VirtualPrecisionClock](expectedDevices, currentDevices)
+		dc := zipVirtualDevicesOfType(MatchVirtualPrecisionClock, exp, cur)
+		deviceChanges = append(deviceChanges, dc...)
+	}
+
+	{
+		exp, cur := selectDevsByType[*vimtypes.VirtualSCSIPassthrough](expectedDevices, currentDevices)
+		dc := zipVirtualDevicesOfType(MatchVirtualSCSIPassthrough, exp, cur)
+		deviceChanges = append(deviceChanges, dc...)
+	}
+
+	{
+		exp, cur := selectDevsByType[*vimtypes.VirtualSerialPort](expectedDevices, currentDevices)
+		dc := zipVirtualDevicesOfType(MatchVirtualSerialPort, exp, cur)
+		deviceChanges = append(deviceChanges, dc...)
+	}
+
+	{
+		exp, cur := selectDevsByType[*vimtypes.VirtualEnsoniq1371](expectedDevices, currentDevices)
+		dc := zipVirtualDevicesOfType(MatchVirtualEnsoniq1371, exp, cur)
+		deviceChanges = append(deviceChanges, dc...)
+	}
+
+	{
+		exp, cur := selectDevsByType[*vimtypes.VirtualHdAudioCard](expectedDevices, currentDevices)
+		dc := zipVirtualDevicesOfType(MatchVirtualHdAudioCard, exp, cur)
+		deviceChanges = append(deviceChanges, dc...)
+	}
+
+	{
+		exp, cur := selectDevsByType[*vimtypes.VirtualSoundBlaster16](expectedDevices, currentDevices)
+		dc := zipVirtualDevicesOfType(MatchVirtualSoundBlaster16, exp, cur)
+		deviceChanges = append(deviceChanges, dc...)
+	}
+
+	{
+		exp, cur := selectDevsByType[*vimtypes.VirtualTPM](expectedDevices, currentDevices)
+		dc := zipVirtualDevicesOfType(MatchVirtualTPM, exp, cur)
+		deviceChanges = append(deviceChanges, dc...)
+	}
+
+	{
+		exp, cur := selectDevsByType[*vimtypes.VirtualWDT](expectedDevices, currentDevices)
+		dc := zipVirtualDevicesOfType(MatchVirtualWDT, exp, cur)
+		deviceChanges = append(deviceChanges, dc...)
+	}
+
+	return deviceChanges
+}
+
+func selectDevsByType[T vimtypes.BaseVirtualDevice](
+	expectedDevices, currentDevices []vimtypes.BaseVirtualDevice) ([]T, []T) {
+
+	return pkgutil.SelectDevicesByType[T](expectedDevices), pkgutil.SelectDevicesByType[T](currentDevices)
+}
+
+func zipVirtualDevicesOfType[T vimtypes.BaseVirtualDevice](
+	matchDevFn func(expectedDev, curDev T) vimtypes.BaseVirtualDevice,
+	expectedDevs, curDevs []T) []vimtypes.BaseVirtualDeviceConfigSpec {
+
+	var deviceChanges []vimtypes.BaseVirtualDeviceConfigSpec //nolint:prealloc
+
+	// For the expected and current devices of a type, zip the devices together to determine
+	// what, if any, edits need to be done. Then add and remove what is leftover.
+	minLen := min(len(expectedDevs), len(curDevs))
+	for idx := range minLen {
+		expectedDev, curDev := expectedDevs[idx], curDevs[idx]
+
+		editDev := matchDevFn(expectedDev, curDev)
+		if editDev != nil {
+			deviceChanges = append(deviceChanges, &vimtypes.VirtualDeviceConfigSpec{
```

Comment:
IIRC, VMCI Device and Video Cards are default devices that can have settable properties on reconfigure -- edits are definitely possible. 
I'm unsure about an 'Edit' on a default device like PointingDevice though. need to verify.

Reply:
The doc at least suggest the the host pointing dev field in the backing can be updated. However, using one of the allowed values from the VirtualPointingDeviceOptions it still fails. There might be something odd going with the field in general because on create valid values from the options appear to be ignore and always ends up as autodetect.

I might just comment out the PointingDevice checks for now since it won't be a very common thing for our usages.